osu.Game/Screens/Select/PlaySongSelect.cs

```diff
@@ -100,14 +110,18 @@ protected override bool OnStart()
 
             Player createPlayer()
             {
+                Player player;
+
                 var replayGeneratingMod = Mods.Value.OfType<ICreateReplayData>().FirstOrDefault();
 
                 if (replayGeneratingMod != null)
-                {
-                    return new ReplayPlayer((beatmap, mods) => replayGeneratingMod.CreateScoreFromReplayData(beatmap, mods));
-                }
+                    player = new ReplayPlayer((beatmap, mods) => replayGeneratingMod.CreateScoreFromReplayData(beatmap, mods));
+                else
+                    player = new SoloPlayer();
+
+                ((IBindableList<ScoreInfo>)player.LeaderboardScores).BindTo(playBeatmapDetailArea.Leaderboard.Scores);
```

Comment:
I think it wouldn't be a problem changing `LeaderboardScores` to `IBindableList<ScoreInfo>`? Just to get rid of this ugly cast.

Reply:
We generally only use `IBindable` for read-only usages, so I'm not sure.

Reply:
Well on `SoloGameplayLeaderboard`, it shouldn't be mutating `LeaderboardScores` I think? It's only using it as a source for score cards.

Reply:
After moving the leaderboard more locally, this is no longer an issue.